File: py/uniphant/worker_context.py
from dataclasses import dataclass
from pathlib import Path
from uuid import UUID

@dataclass(frozen=True)
class WorkerContext:
    foreground: bool
    host_id: UUID
    host_id_file: Path
    host_name: str
    process_id: UUID
    root_dir: Path
    script_dir: Path
    secret_dir: Path
    secrets_root: Path
    worker_id: UUID
    worker_type: str

File: rust/src/worker_context.rs
use std::path::PathBuf;
use uuid::Uuid;

#[derive(Debug, PartialEq, Eq, Hash, Clone)]
pub struct WorkerContext {
    pub foreground: bool,
    pub host_id: Uuid,
    pub host_id_file: PathBuf,
    pub host_name: String,
    pub process_id: Uuid,
    pub root_dir: PathBuf,
    pub secret_dir: PathBuf,
    pub secrets_root: PathBuf,
    pub worker_dir: PathBuf,
    pub worker_id: Uuid,
    pub worker_type: String,
}
